Compiler/HandlebarsGrammar.peg

```diff
@@ -153,6 +154,16 @@ UnlessClosingToken
 ElseToken
   = Opening ElseKeyword Closing
 
+//#Equals
+EqualsBlock <EqualsBlock>
+  = expr:EqualsOpeningToken body:BlockBody ElseToken elseBlock:BlockBody EqualsClosingToken { new EqualsBlock(expr, elseBlock, body, exprStart.Line, exprStart.Column) }
+  / expr:EqualsOpeningToken body:BlockBody EqualsClosingToken { new EqualsBlock(expr, body, exprStart.Line, exprStart.Column) }
+
+EqualsOpeningToken <Expression>
+  = Opening BlockOpeningIdentifier EqualsKeyword expr:Expression Closing { expr }
```

Comment:
You need to return two Expressions here for the comparison.
I would propose something like:
```
EqualsOpeningToken <EqualsExpression>
  = Opening BlockOpeningIdentifier EqualsKeyword lhs:Expression rhs:Expression Closing { new EqualsExpression(lhs, rhs)  } 
```